cloud/gc_dns.py

```diff
@@ -0,0 +1,349 @@
+  name:
+    description:
+      - the zone name identifier
+    required: false
+    aliases: ["zone_name", "zone"]
+  record:
+    description:
+      - the full DNS record to create or delete
+    required: false
+    default: null
+    aliases: []
+  ttl:
+    description:
+      - the time-to-live in seconds to cache this record with DNS servers
+    required: false
+    default: null
+    aliases: []
+  record_type:
+    description:
+      - the type of resource record, e.g. A, AAAA, CNAME, MX, etc
+    required: false
+    default: null
+    aliases: []
+    choices: ['A','AAAA','CNAME','MX','NS','PTR','SOA','SPF','SRV','TXT']
+  value:
+    description:
+      - comma-separated list of appropriate record data (e.g. IP, DNS name)
+    required: false
+    default: null
+    aliases: []
+  pem_file:
+    description:
+      - path to the pem file associated with the service account email
+    required: false
+    default: null
+    aliases: []
+  project_id:
+    description:
+      - your GCE project ID
+    required: false
+    default: null
+    aliases: []
+  service_account_email:
+    description:
+      - service account email
+    required: false
+    default: null
+    aliases: []
+
+requirements: [ "libcloud" ]
+author: Eric Johnson <erjohnso@google.com>
+'''
+
+EXAMPLES = '''
+# List existing managed zones
+- local_action:
+    module: gc_dns
+    command: list
+
+# List defined records for the specified  managed zone
+- local_action:
+    module: gc_dns
+    command: list
+    name: my-example-zone
+
+# Basic example of creating a new managed zone
+- local_action:
+    module: gc_dns
+    command: create
+    name: my-example-zone
+    dns_name: example.com.
+    description: My first managed zone
+
+# Basic example of deleting a managed zone
+- local_action:
+    module: gc_dns
+    command: delete
+    name: my-example-zone
+
+# Basic example of adding two 'A' records to an existing zone
+- local_action:
+    module: gc_dns
+    command: create
+    zone: my-example-zone
+    record: foo.example.com.
+    value: 123.45.67.89,98.76.54.32
+    record_type: A
+    ttl: 7200
+
+# Basic example of deleting an existing record
+- local_action:
+    module: gc_dns
+    command: delete
+    zone: my-example-zone
+    record: db.example.com.
+    record_type: A
+    value: 123.45.67.89
+'''
+
+import sys
+
+try:
+    from libcloud.dns.types import (Provider, ZoneDoesNotExistError,
+                                    RecordDoesNotExistError)
+    from libcloud.dns.providers import get_driver
+    from libcloud.common.google import (GoogleBaseAuthConnection,
+                                        ResourceExistsError,
+                                        GoogleInstalledAppAuthConnection,
+                                        GoogleBaseConnection)
+    _ = Provider.GOOGLE
+except ImportError:
+    print("failed=True " + \
+        "msg='libcloud with GCE support >=0.15.0 required for this module'")
+    sys.exit(1)
```

Comment:
use fail_json from module instead, we normally defer the error until module is instantiated to use this facility